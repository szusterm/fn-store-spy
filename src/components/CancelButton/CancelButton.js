import React, {Component} from 'react';
import PropTypes from 'prop-types';

import './styles.scss';

export class CancelButton extends Component {
	render() {
		const {onClick} = this.props;

		return (
			<button
				className={'cancel-button'}
				onClick={onClick}
				aria-label={'Cancel'}
			>
				<span className={'material-icons'}>close</span>
			</button>
		);
	}
}

CancelButton.propTypes = {
	onClick: PropTypes.func //redux
};

export default CancelButton;