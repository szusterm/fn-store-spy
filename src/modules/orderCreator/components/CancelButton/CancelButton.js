import React, {Component} from 'react';
import PropTypes from 'prop-types';
import {connect} from 'react-redux';

import {closeOrderList} from '../../actions';

import './styles.scss';

export class CancelButton extends Component {
	render() {
		const {closeOrderList} = this.props;

		return (
			<div className={'col-8'}>
				<button
					className={'cancel-button'}
					onClick={closeOrderList}
				>
					<span className={'material-icons'}>close</span>
				</button>
			</div>
		);
	}
}

CancelButton.propTypes = {
	closeOrderList: PropTypes.func //redux
};

const mapDispatchToProps = {closeOrderList};

export default connect(null, mapDispatchToProps)(CancelButton);