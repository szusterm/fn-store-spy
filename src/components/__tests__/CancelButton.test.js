import React from 'react';
import {shallow} from 'enzyme';

import {CancelButton} from '../CancelButton/CancelButton';

const setup = (propOverrides) => {
	const props = Object.assign({
		onClick: () => true //redux
	}, propOverrides);

	const wrapper = shallow(<CancelButton {...props}/>);

	return {props, wrapper};
};

describe('CancelButton Component', () => {
	it('renders correctly', () => {
		const {wrapper} = setup();

		expect(wrapper).toMatchSnapshot();
	});

	it('calls onClick() prop after click the button', () => {
		const {wrapper, props} = setup({
			onClick: jest.fn()
		});

		wrapper.find('button').simulate('click');

		expect(props.onClick).toHaveBeenCalledTimes(1);
	});
});