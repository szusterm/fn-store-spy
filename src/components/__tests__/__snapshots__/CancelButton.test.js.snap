// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CancelButton Component renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CancelButton
    onClick={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <span
        className="material-icons"
      >
        close
      </span>,
      "className": "cancel-button",
      "onClick": [Function],
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "close",
        "className": "material-icons",
      },
      "ref": null,
      "rendered": "close",
      "type": "span",
    },
    "type": "button",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <span
          className="material-icons"
        >
          close
        </span>,
        "className": "cancel-button",
        "onClick": [Function],
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "close",
          "className": "material-icons",
        },
        "ref": null,
        "rendered": "close",
        "type": "span",
      },
      "type": "button",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
